src/qz/ws/PrintSocketServer.java

```diff
@@ -223,12 +225,37 @@ public static void main(String ... args) {
         App.main(args);
     }
 
+    /**
+     * Parses WebSocket ports from preferences or fallback to defaults is a problem is found
+     */
+    public static void parseWebSocketPorts() {
+        Integer[] secure = PrefsSearch.getIntegerArray(ArgValue.WEBSOCKET_SECURE_PORTS);
+        Integer[] insecure = PrefsSearch.getIntegerArray(ArgValue.WEBSOCKET_INSECURE_PORTS);
+        boolean fallback = false;
+        if(secure.length == 0 || insecure.length == 0) {
+            log.warn("One or more WebSocket ports is empty, falling back to defaults");
+            fallback = true;
+        }
+        if(secure.length != insecure.length) {
+            log.warn("Secure ({}) and insecure ({}) WebSocket port counts mismatch, falling back to defaults", secure, insecure);
+            fallback = true;
+        }
+        if(fallback) {
+            log.warn("Falling back to default WebSocket ports: ({}), ({})", secure, insecure);
+            secure = Constants.DEFAULT_WSS_PORTS;
+            insecure = Constants.DEFAULT_WS_PORTS;
+        }
+
+        securePorts = Collections.unmodifiableList(Arrays.asList(secure));
+        insecurePorts = Collections.unmodifiableList(Arrays.asList(insecure));
```

Comment:
If we look back at the path this array takes, we can see back in `getIntegerArray(ArgValue argValue, Properties ... propsArray)` it is constructed as a list, then converted to an array. Could we just keep it as a List?

Reply:
Well, we still have to construct it due to the Constants being primitive, but done with the larger refactor that was just pushed.